modules/yugabytedb/yugabytedb.go

```diff
@@ -0,0 +1,249 @@
+package yugabytedb
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"strings"
+
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/wait"
+	"github.com/yugabyte/gocql"
+)
+
+const (
+	mastDashboardPort    = "7000"
+	tServerDashboardPort = "9000"
+)
+
+const (
+	ycqlPort = "9042/tcp"
+
+	ycqlKeyspaceEnv = "YCQL_KEYSPACE"
+	ycqlUserNameEnv = "YCQL_USER"
+	ycqlPasswordEnv = "YCQL_PASSWORD"
+
+	ycqlKeyspace = "yugabyte"
+	ycqlUserName = "yugabyte"
+	ycqlPassword = "yugabyte"
+)
+
+const (
+	ysqlPort = "5433/tcp"
+
+	ysqlDatabaseNameEnv     = "YSQL_DB"
+	ysqlDatabaseUserEnv     = "YSQL_USER"
+	ysqlDatabasePasswordEnv = "YSQL_PASSWORD"
+
+	ysqlDatabaseName     = "yugabyte"
+	ysqlDatabaseUser     = "yugabyte"
+	ysqlDatabasePassword = "yugabyte"
+)
+
+// YugabyteDBContainer represents the yugabyteDB container type used in the module
+type YugabyteDBContainer struct {
+	testcontainers.Container
+
+	ysqlDatabaseName     string
+	ysqlDatabaseUser     string
+	ysqlDatabasePassword string
+}
+
+// Run creates an instance of the yugabyteDB container type and automatically starts it.
+// A default configuration is used for the container, but it can be customized using the
+// provided options.
+// When using default configuration values it is recommended to use the provided
+// [*YugabyteDBContainer.YSQLConnectionString] and [*YugabyteDBContainer.YCQLConfigureClusterConfig]
+// methods to use the container in their respective clients.
+//
+// # Example using YSQL dialect:
```

Comment:
question: I'm always nervous about examples in comments as they can be broken or go stale without people noticing. We also have actual examples, so would be tempted to say lets remove these. 

@mdelapenya what do you think?

Reply:
Looked around and it seems that this is indeed not a common practice in the repository. Happy to remove if you want.

Reply:
I'd prefer having a testable example demonstrating how to use it. Wdyt?

Reply:
Agreed, looks like we have examples which cover these. If there are any missing just add concrete examples and remvoe these from the comments.

Reply:
✅ Attended